PocketKit/Sources/PocketKit/BannerPresenter.swift

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Textile
+import Combine
+import SwiftUI
+
+class BannerPresenter: ObservableObject {
+    var bannerData: BannerModifier.BannerData?
+    @Published var shouldPresentBanner = false
+
+    private let notificationCenter: NotificationCenter
+    private var subscriptions: Set<AnyCancellable> = []
+
+    init(notificationCenter: NotificationCenter) {
+        self.notificationCenter = notificationCenter
+    }
+
+    func listen() {
+        notificationCenter.publisher(for: .bannerRequested).sink { [weak self] notification in
```

Comment:
Can we add an explicit main thread listener? This could be posted from another one.

Reply:
yeah, we should. i think SwiftUI expects things to be published on the main thread, so i think if we receive on the main thread, and set on the main thread, we should be okay.

Reply:
Yea the main thing i care about is receiving on the main thread